src/GafferScene/RenderController.cpp

```diff
@@ -519,6 +522,25 @@ class RenderController::SceneGraph
 		{
 			if( !m_attributesInterface )
 			{
+				if( !m_fullAttributesSubstituted )
```

Comment:
I'm not sure it's right to substitute the attributes in place like this, is it? Won't child locations end up inheriting the already-substituted attributes, and then not be able to apply their own, more-specific, substitutions with their own attribute values? I think the only thing we can do here is make a temporary substituted attributes and then throw it away (part of the reason I was interested in caching AttributeInterfaces).

I think this also implies we don't need the extra `m_fullAttributesSubstituted` state, which would be nice, because the logic/state in this class is already pretty complex.